src/jit/compiler.cpp

```diff
@@ -11097,3 +11097,831 @@ bool Compiler::killGCRefs(GenTree* tree)
+
+    return m_VariableLifeBarrier->hasLiveRangesToDump();
+}
+
+// Reset the barrier so as to dump only next block changes on next block
+void VariableLiveDescriptor::endBlockLiveRanges()
+{
+    noway_assert(m_VariableLifeBarrier != nullptr);
+
+    // make "m_VariableLifeBarrier->m_StartingLiveRange" now points to nullptr for printing purposes
+    m_VariableLifeBarrier->resetDumper(m_VariableLiveRanges);
+}
+
+#endif // DEBUG
+
+//------------------------------------------------------------------------
+//                      VariableLiveKeeper
+//------------------------------------------------------------------------
+
+void Compiler::initializeVariableLiveKeeper()
+{
+    CompAllocator allocator = getAllocator(CMK_VariableLiveRanges);
+
+    int amountTrackedVariables = opts.compDbgInfo ? info.compLocalsCount : 0;
+    int amountTrackedArgs      = opts.compDbgInfo ? info.compArgsCount : 0;
+
+    varLiveKeeper = allocator.allocate<VariableLiveKeeper>(1);
+    memset(varLiveKeeper, 0, sizeof(*varLiveKeeper));
+    new (varLiveKeeper, jitstd::placement_t()) VariableLiveKeeper(amountTrackedVariables, amountTrackedArgs, this);
+}
+
+VariableLiveKeeper* Compiler::getVariableLiveKeeper() const
+{
+    return varLiveKeeper;
+};
+
+//------------------------------------------------------------------------
+// VariableLiveKeeper: Create an instance of the object in charge of managing
+//  VariableLiveRanges and intialize the array "lvaLiveDsc".
+//
+// Arguments:
+//    totalLocalCount   - the amount of args, special args and IL Local
+//      variables in the method.
+//    argsCount         - the amount of args and special args in the method.
+//    compiler          - a compiler instance
+//
+VariableLiveKeeper::VariableLiveKeeper(unsigned int totalLocalCount, unsigned int argsCount, Compiler* comp)
+{
+    // Fill internal properties
+    lvLiveDscCount              = totalLocalCount;
+    lvLiveArgsCount             = argsCount;
+    compiler                    = comp;
+    lastBasicBlockHasBeenEmited = false;
+
+    if (0 < lvLiveDscCount)
+    {
+        // Get the allocator used for all the VariableLiveRange objects
+        CompAllocator allocator = compiler->getAllocator(CMK_VariableLiveRanges);
+
+        // Allocate memory for "lvaLiveDsc" and initialize each "VariableLiveDescriptor"
+        lvaLiveDsc            = allocator.allocate<VariableLiveDescriptor>(lvLiveDscCount);
+        size_t lvaLiveDscSize = sizeof(*lvaLiveDsc);
+        memset(lvaLiveDsc, 0, lvaLiveDscSize);
+
+        unsigned int            varNum;
+        VariableLiveDescriptor* varLiveDsc;
+        for (varNum = 0, varLiveDsc = lvaLiveDsc; varNum < lvLiveDscCount; varNum++, varLiveDsc++)
+        {
+            new (varLiveDsc, jitstd::placement_t()) VariableLiveDescriptor(allocator);
+        }
+    }
+}
+
+VariableLiveKeeper::~VariableLiveKeeper()
+{
+    // Iterates all the VariableLiveDescriptor and clal its destructor
```

Comment:
Shouldn't we just be able to do `delete[] lvaLiveDsc`?

Reply:
Just delete the whole thing